src/Forms/GridField/GridFieldFilterHeader.php

```diff
@@ -55,10 +75,16 @@ public function getURLHandlers($gridField)
 
     /**
      * @param bool $useLegacy
+     * @param callable|null $updateSearchForm This will be removed in 5.0
      */
-    public function __construct($useLegacy = false)
-    {
+    public function __construct(
+        $useLegacy = false,
+        callable $updateSearchContext = null,
+        callable $updateSearchForm = null
+    ) {
         $this->useLegacyFilterHeader = $useLegacy;
+        $this->updateSearchContextCallback = $updateSearchContext;
+        $this->updateSearchFormCallback = $updateSearchForm;
```

Comment:
I think it'd be good to see getters and setters for these props, then use them in this class

Reply:
I _think_ they were omitted because they're intended to be removed in 5.0. Best to reduce the API surface considering that?

Reply:
If that's the case we should mark them as deprecated